src/RouteCollectionInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Router;
+
+interface RouteCollectionInterface
+{
+    /**
+     * @return array
+     */
+    public function getRoutes(): array;
+
+    /**
+     * @param string $name
+     * @return Route
+     */
+    public function getRoute(string $name): Route;
+
+    /**
+     * Returns routes tree array
+     *
+     * @return array
+     */
+    public function getRouteTree(): array;
```

Comment:
It's not a good idea to have `array` as return type in an interface in case certain structure of an array is assumed.